config/database.yml

```diff
@@ -58,7 +74,15 @@ development:
 # Do not set this db to the same as development or production.
 test:
   <<: *default
+  # Try to set the database to the value specified in DATABASE_URL if we're
+  # actually in the expected environment. Otherwise, default to
+  # web-monitoring-db_test
+  <% if ENV['RAILS_ENV'] == 'test' && ENV['DATABASE_URL'] %>
+  <% db_uri = URI.parse(ENV['DATABASE_URL']) %>
+  database: <%= db_uri.path || web-monitoring-db_test %>
+  <% else %>
   database: web-monitoring-db_test
+  <% end %>
```

Comment:
This doesn’t work if I’m in a dev environment and I’m running a command like `rake db:schema:load` and my test DB has a different user from my dev DB or it has a different name. (When the rake command flips to repeat in the test environment, it’s picking up this config, but the users from my `DATABASE_URL`, and it gets messy.)